### DISCUSSION TITLE
Nuxt slow down to every new user

### DISCUSSION THREAD

**Discussion by giosh** [nuxt/nuxt#9423] [Questions (Nuxt 2)]
We’re stuck on a problem. We have to put online the frontend of an ecommerce developed in Nuxt, but at the moment we aren't confortable, because by doing traffic tests we see that the response times of the website slow down to every new user. We know that the visits will be more than 10K month and the contemporary traffic can reach up to 250 people.

This is our infrastructure:
EC2 server on AWS with 4 Cpus and 8GB of Ram
OS Ubuntu 20 LTS Apache web server + mpm_event
Cache Server Redis
Mongodb database 4.4
Service API (NODEJS + Express) Port :5000 (Outside Nuxt) Nuxt Port :3000 – Reverse Proxy

We also also has performance issues on a fresh install project that launches a single API in SSR (via nuxt-Axios). As SSR API endpoint we aim for the localhost:5000

In support of the ecommerce we developed a CMS in PHP that uses the same APIs. By testing PHP page with the same requests, we don’t experience the performance problems we see on Nuxt. In addition, the bandwidth that occupies the test run on PHP is much lower than when we run the test on Nuxt.

Anyone with the same problem? Anyone with a solution? Thank you!

**Comment by TheAlexLichter** [comment]
Improving performance is a broad topic so I can't go into every detail in this answer.

The main part that comes to my mind is that SSR calls are synchronous and will block the event loop. The more concurrent requests you will have on the *same* server, the more time it will take. This can be mitigated by using a cluster (multiple instances).

BTW: This is a conceptual "problem" of SSR, not strictily related to Nuxt itself.

**Comment by kissu** [comment]
[Your question](https://stackoverflow.com/q/67925878/8816585) got closed on StackOverflow for a reason (too broad essentially).
Asking it here will not help much: you're listing a lot of versions and hardware while your issue could be code-wise.
Try to benchmark and analyze your code, your requests, maybe try it on a more bulky local hardware and scope it to more specific parts.
As of right now, nobody will be able to point an **exact** reason of why this is slow.
